quickwit/quickwit-doc-mapper/src/default_doc_mapper/field_mapping_entry.rs

```diff
@@ -103,6 +109,115 @@ impl Default for QuickwitNumericOptions {
     }
 }
 
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq, utoipa::ToSchema)]
+#[serde(deny_unknown_fields)]
+pub struct QuickwitBytesOptions {
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub description: Option<String>,
+    #[serde(default = "default_as_true")]
+    pub stored: bool,
+    #[serde(default = "default_as_true")]
+    pub indexed: bool,
+    #[serde(default)]
+    pub fast: bool,
+    #[serde(default)]
+    pub input_format: BinaryFormat,
+    #[serde(default)]
+    pub output_format: BinaryFormat,
+}
+
+impl Default for QuickwitBytesOptions {
+    fn default() -> Self {
+        Self {
+            description: None,
+            indexed: true,
+            stored: true,
+            fast: false,
+            input_format: BinaryFormat::default(),
+            output_format: BinaryFormat::default(),
+        }
+    }
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Hash, Default)]
+pub enum BinaryFormat {
+    #[default]
+    Base64,
+    Hex,
+}
+
+impl BinaryFormat {
+    pub fn as_str(&self) -> &str {
+        match self {
+            BinaryFormat::Base64 => "base64",
+            BinaryFormat::Hex => "hex",
+        }
+    }
+
+    pub fn format_to_json(&self, value: &[u8]) -> JsonValue {
+        match self {
+            BinaryFormat::Base64 => BASE64_STANDARD.encode(value).into(),
+            BinaryFormat::Hex => hex::encode(value).into(),
+        }
+    }
+
+    pub fn parse_json(&self, json_val: JsonValue) -> Result<TantivyValue, String> {
+        let byte_str = if let JsonValue::String(byte_str) = json_val {
+            byte_str
+        } else {
+            return Err(format!(
+                "Expected {} string, got `{json_val}`.",
+                self.as_str()
+            ));
+        };
+        let payload = match self {
+            BinaryFormat::Base64 => {
+                BASE64_STANDARD
+                    .decode(&byte_str)
+                    .map_err(|base64_decode_err| {
+                        format!("Expected base64 string, got `{byte_str}`: {base64_decode_err}")
+                    })?
+            }
+            BinaryFormat::Hex => hex::decode(&byte_str).map_err(|hex_decode_err| {
+                format!("Expected hex string, got `{byte_str}`: {hex_decode_err}")
+            })?,
+        };
+        Ok(TantivyValue::Bytes(payload))
+    }
+}
+
+impl FromStr for BinaryFormat {
+    type Err = String;
+
+    fn from_str(bytes_format_str: &str) -> Result<Self, Self::Err> {
+        let bytes_format = match bytes_format_str.to_lowercase().as_str() {
+            "base64" => BinaryFormat::Base64,
+            "hex" => BinaryFormat::Hex,
+            _ => {
+                return Err(format!("Unknown byte format: `{bytes_format_str}`."));
+            }
+        };
+        Ok(bytes_format)
+    }
+}
+
+impl Serialize for BinaryFormat {
```

Comment:
Can't we derive `Serialize` and `Deserialize` with `#[serde(rename_all = "snake_case")]` instead?